### ISSUE TITLE
DerObjectId doesn't like large second or third digits

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [Legrandin/pycryptodome#739]
```python
from Crypto.Util.asn1 import DerObjectId

_OID_ARGON2 = '2.99.42'

obj1 = DerObjectId(_OID_ARGON2)
print('Encoded object (it thinks it represents %s):\n%s' % (repr(obj.value), obj.encode().hex()))
# Encoded object (it thinks it represents '2.99.42'):
# 0602b32a

obj2 = DerObjectId().decode(obj.encode())
print('Decoded and re-encoded object (it thinks it represents %s)\n%s' % (repr(obj2.value), obj2.encode().hex()))
# Decoded and re-encoded object (it thinks it represents '4.19.42')
# 0602b32a
```

I also tried:

- `1.2.42` (OK)
- `2.2.42` (OK)
- `1.99.42` (FAIL)
- `2.42.99` (FAIL)
- `2.2.99` (OK)
- `2.2.999` (OK)
- `1.3.6.1.4.1.7606.2291.138023.36282760082` (OK)

Per that last case, it looks like it's not stumbling over total length, but rather only over the first few digits.

**Comment by JamesTheAwesomeDude** [comment]
I just realized I wasn't trying in the real test range; when I try `2.999.42`, it actually *raises* an error--

```
  File "c:\users\edingtonjames\source\repos\pycryptodome\lib\Crypto\Util\asn1.py", line 855, in encode
    self.payload = bchr(40*comps[0]+comps[1])
  File "c:\users\edingtonjames\source\repos\pycryptodome\lib\Crypto\Util\py3compat.py", line 116, in bchr
    return bytes([s])
ValueError: bytes must be in range(0, 256)
```
https://github.com/sop/asn1/issues/9#issuecomment-469652080

> The "0..39" limitation applies only to the top arcs 0 and 1, **not to 2.** \[ITU-T Rec. X.960 §8.19.4\]

**Comment by Legrandin** [comment]
Thanks, fixed with 09f0ac570c6a47984c692eb82695d493de971117

**Comment by Legrandin** [comment]
Fixed in v3.18.0
